src/compressed_tensors/compressors/model_compressor.py

```diff
@@ -170,35 +184,44 @@ def from_pretrained_model(
         )
 
     @staticmethod
-    def parse_sparsity_config(compression_config: Dict) -> Union[Dict, None]:
+    def parse_sparsity_config(
+        compression_config: Union[Dict[str, Any], "CompressedTensorsConfig"]
+    ) -> Union[Dict[str, Any], None]:
+        """
+        Parse sparsity config from quantization/compression config. Sparsity
+        config is nested inside q/c config
+
+        :param compression_config: quantization/compression config
+        :return: sparsity config
+        """
         if compression_config is None:
             return None
-        if SPARSITY_CONFIG_NAME not in compression_config:
-            return None
-        if hasattr(compression_config, SPARSITY_CONFIG_NAME):
-            # for loaded HFQuantizer config
-            return getattr(compression_config, SPARSITY_CONFIG_NAME)
-        if SPARSITY_CONFIG_NAME in compression_config:
-            # for loaded HFQuantizer config from dict
-            return compression_config[SPARSITY_CONFIG_NAME]
-
-        # SparseAutoModel format
+
+        if is_compressed_tensors_config(compression_config):
+            s_config = compression_config.sparsity_config
+            return s_config.dict() if s_config is not None else None
+
         return compression_config.get(SPARSITY_CONFIG_NAME, None)
 
     @staticmethod
-    def parse_quantization_config(compression_config: Dict) -> Union[Dict, None]:
+    def parse_quantization_config(
+        compression_config: Union[Dict[str, Any], "CompressedTensorsConfig"]
+    ) -> Union[Dict[str, Any], None]:
+        """
+        Parse quantization config from quantization/compression config. The
+        quantization are all the fields that are not the sparsity config or
+        metadata fields
+
+        :param compression_config: quantization/compression config
+        :return: quantization config without sparsity config or metadata fields
+        """
         if compression_config is None:
             return None
 
-        if hasattr(compression_config, QUANTIZATION_CONFIG_NAME):
-            # for loaded HFQuantizer config
-            return getattr(compression_config, QUANTIZATION_CONFIG_NAME)
+        if is_compressed_tensors_config(compression_config):
```

Comment:
leave a quick comment indicating that this is for HFQuant

Reply:
Would a comment in the function itself suffice? We'll probably be using this function a decent amount in the future it's already used 3 times in this PR